lib/ruby_lsp/test_helper.rb

```diff
@@ -76,5 +77,16 @@ def pop_message(outgoing_queue, &block)
       message = outgoing_queue.pop until block.call(message)
       message
     end
+
+    #: (IO output) -> Array[Hash[Symbol, untyped]]
+    def parse_json_api_stream(output)
```

Comment:
If we're going to provide this as an API for all add-ons, would it be more helpful if we could instead provide a general API for `read_message` that reads a single JSON RPC message from a given IO object?

Reply:
Maybe. There might be slightly different needs for something used as part of the production code vs something only used by the tests. Let's do this test part first then re-consider?

Reply:
In that case, let's put this in our test helper and not in the API exposed to add-ons.

However, we should be opinionated about how people test add-ons and the closest to how it works in prod the better.